src/sentry/notifications/notificationcontroller.py

```diff
@@ -0,0 +1,460 @@
+        ],
+    ]:
+        """
+        Returns a mapping of all enabled notification setting providers for the enabled options.
+        Note that this includes default settings for any notification types that are not set.
+        """
+
+        setting_options = self._get_layered_setting_options(**kwargs)
+        setting_providers = self._get_layered_setting_providers(**kwargs)
+
+        setting_option_and_providers: MutableMapping[
+            Recipient,
+            MutableMapping[
+                Scope,
+                MutableMapping[
+                    NotificationSettingEnum,
+                    MutableMapping[ExternalProviderEnum, NotificationSettingsOptionEnum],
+                ],
+            ],
+        ] = defaultdict(lambda: defaultdict(lambda: defaultdict(dict)))
+        for recipient, setting_option in setting_options.items():
+            for scope, setting in setting_option.items():
+                for type, option_value in setting.items():
+                    if option_value == NotificationSettingsOptionEnum.NEVER:
+                        continue
+
+                    recipient_providers = setting_providers[recipient][scope][type]
+                    for provider, provider_value in recipient_providers.items():
+                        if provider_value == NotificationSettingsOptionEnum.NEVER:
+                            continue
+
+                        setting_option_and_providers[recipient][scope][type][
+                            provider
+                        ] = provider_value
+
+        return setting_option_and_providers
+
+    def get_settings_options_for_user_by_projects(
+        self, user: Recipient
+    ) -> MutableMapping[
+        int,
+        MutableMapping[
+            NotificationSettingEnum,
+            MutableMapping[ExternalProviderEnum, NotificationSettingsOptionEnum],
+        ],
+    ]:
+        """
+        Returns a mapping of project IDs to enabled notification settings for the given user.
+        """
+        if not self.project_ids:
+            raise Exception("Must specify project_ids")
+
+        notification_settings = self.get_all_enabled_settings()
+        user_settings = notification_settings[user]
+
+        result = {}
+        for scope, setting in user_settings.items():
+            (scope_type, scope_identifier) = scope
+            if scope_type != NotificationScopeEnum.PROJECT:
+                continue
+
+            if not isinstance(scope_identifier, int) or scope_identifier not in self.project_ids:
+                continue
+
+            result[scope_identifier] = setting
+
+        return result
+
+    def get_subscriptions_status_for_projects(
+        self, user: Recipient
+    ) -> Mapping[int, Tuple[bool, bool]]:
+        """
+        Returns whether the user is subscribed for each project, and the subscription object if it exists.
+        {project_id -> (is_disabled, is_active, subscription object)}
+        """
+        if not self.type:
```

Comment:
@snigdhas should this be an optional input to this function and we default to `self.type`?